.github/workflows/repro-check.yml

```diff
@@ -0,0 +1,60 @@
+name: Repro check
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      github_sha:
+        description: Commit on which the repro check should be run
+        required: true
+        type: string
+  # Used as a reusable workflow within release-testing workflow
+  workflow_call:
+    inputs:
+      github_sha:
+        description: Commit on which the repro check should be run
+        required: true
+        type: string
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  repro-check:
+    name: Run repro-check for ${{ inputs.github_sha }}
+    runs-on:
+      labels: dind-large
+    container:
+      image: ubuntu:24.04
+      options: >-
+        -e NODE_NAME --privileged --cgroupns host
+    steps:
+      - name: Setup podman
+        shell: bash
+        run: |
+          apt-get update
+          apt-get -y upgrade
+          apt-get -y install curl sudo git podman
+
+          # Create a new non-root user with home directory
+          useradd -m -s /bin/bash ciuser
+          
+          # Create a passwordless sudo rule for the current user (usually root in containers, but safe to include)
+          # Required because the repro check uses it.
+          echo "ciuser ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/nopasswd
+          chmod 0440 /etc/sudoers.d/nopasswd
```

Comment:
Can't we use the preexisting `ubuntu` user here?

Note that most external reviewers will run repro-check on an Ubuntu EC2 instance where they're by default the `ubuntu` user. In the past we actually ran into an issue with that (I can look it up if desired) so it would be good to test for that default case.

Reply:
In the docker image there is no `ubuntu` user. We only have a root user out of the box. Its because it is used as a base to most of the other images I guess. I can rename the user to be `ubuntu` user if needed. But at the end, the `repro-check` script will run our dev docker image which will have a `ubuntu` user there as expected.

Reply:
>In the docker image there is no ubuntu user. 

Are you sure? What about:
```
$ podman run -it ubuntu:24.04 cat /etc/passwd | grep ubuntu
ubuntu:x:1000:1000:Ubuntu:/home/ubuntu:/bin/bash
```

Reply:
Yeah I think using the `ubuntu` is better since that's what reviewers use as well. I looked up the previous time repro-check failed for external reviewers due to them using the `ubuntu` user. See the [slack thread](https://dfinity.slack.com/archives/CL7Q2RXUM/p1739802792503969?thread_ts=1739783081.578219&cid=CL7Q2RXUM).